### PR TITLE
Fix dataclass kernel argument containing list raises std::bad_cast

### LINKED ISSUE #4846: [python] Dataclass kernel argument with a list[int] field raises std::bad_cast when the list contains a negative value
### Required prerequisites

- [x] Consult the [security policy](https://github.com/NVIDIA/cuda-quantum/security/policy). If reporting a security vulnerability, do not report the bug using this form. Use the process described in the policy to report the issue.
- [x] Make sure you've read the [documentation](https://nvidia.github.io/cuda-quantum/latest). Your issue may be addressed there.
- [x] Search the [issue tracker](https://github.com/NVIDIA/cuda-quantum/issues) to verify that this hasn't already been reported. +1 or comment there if it has.

### Describe the bug

A `@dataclass` used as a kernel argument works when its `list[int]` fields
contain only non-negative values, but launching raises
`RuntimeError: std::bad_cast` as soon as any list element is negative. The
same list passed as a bare `list[int]` argument works fine with negative
values, so the defect is specific to the struct-member marshaling path (it
behaves as if elements are cast through an unsigned type during packing).

Observed matrix:

| Case | Result |
|---|---|
| dataclass, scalar fields only | OK |
| dataclass, `list[int]`/`list[float]` fields, non-negative values | OK (tested 1-5 list fields, mixed lengths) |
| dataclass, any `list[int]` field containing a negative value | `std::bad_cast` |
| bare `list[int]` argument containing negative values | OK |
| captured dataclass with negative list element | `std::bad_cast` |

### Steps to reproduce the bug

```python
from dataclasses import dataclass
import cudaq

cudaq.set_target("qpp-cpu")

@dataclass(slots=True)
class Args:
    values: list[int]

@cudaq.kernel
def kernel(args: Args):
    q = cudaq.qvector(1)
    if args.values[0] < 0:
        x(q[0])

cudaq.sample(kernel, Args([1]))    # OK
cudaq.sample(kernel, Args([-1]))   # RuntimeError: std::bad_cast

@cudaq.kernel
def bare(values: list[int]):
    q = cudaq.qvector(1)
    if values[0] < 0:
        x(q[0])

cudaq.sample(bare, [-1])           # OK -> '1'
```

### Expected behavior

Negative integers in a dataclass `list[int]` field marshal identically to
negative integers in a bare `list[int]` argument.

### Is this a regression? If it is, put the last known working version (or commit) here.

Not a regression

### Environment

- CUDA-Q built from source at commit `0be565550f4c23affdcbed9e4eaec38d2d0915e6`
- Python 3.11.x, target `qpp-cpu` (library mode)
- AlmaLinux 8.10, x86_64

All reproducers below were executed and verified against this build.

### PR DESCRIPTION
**Pull request by atgeller**
Reuse vector marshaling for struct marshaling.
Changes `handleStructMemberVariable` to delegate to `handleVectorElements` instead of having it's own incorrect vector handling logic. Add a regression test for negative numbers in lists in structs.

Fixes #4846.

### CODE REVIEW

File: python/runtime/cudaq/platform/py_alt_launch_kernel.cpp [hunk 1]
```diff
@@ -199,54 +199,26 @@ void cudaq::handleStructMemberVariable(void *data, std::size_t offset,
...
+                "Type not supported for custom struct in kernel.");
+
+        // Build the std::vector<T> (with the correct element type, including
+        // nested vectors) via the shared element handler, then copy its header
+        // into the struct.
+        auto asList = nanobind::cast<nanobind::list>(value);
+        void *values = handleVectorElements<style>(ty.getElementType(), asList);
+        // synthesis path stores a span {ptr, size_t}; argsCreator stores the
+        // full std::vector header {ptr, ptr, ptr}. The copied size does not
+        // depend on the element type.
```
Comment (sacpis) [inline python/runtime/cudaq/platform/py_alt_launch_kernel.cpp]: nit: Does this comment (`The copied size does not depend on the element type.`) apply to ` std::vector<bool>` type?

**Review by sacpis — APPROVED** [review]
LGTM. Thanks @atgeller.

File: python/runtime/cudaq/platform/py_alt_launch_kernel.cpp (hunk 1 above)
Comment (atgeller) [reply python/runtime/cudaq/platform/py_alt_launch_kernel.cpp]: It does not. Let me make sure that gets handled correctly.

Comment (atgeller) [reply python/runtime/cudaq/platform/py_alt_launch_kernel.cpp]: It does not. I fixed it and added a test case.
